### PR TITLE
Add abort-on-container-exit option to compose.up

### PR DESCRIPTION
**Pull request by anesabml**
Related issue: #166

### CODE REVIEW

**Comment by anesabml** [comment]
@gabrieldemarmiesse any idea why the flake8 command fails?

**Comment by gabrieldemarmiesse** [comment]
Going at the bottom of the logs, a bit hidden, I see
```
./python_on_whales/components/compose/cli_wrapper.py:267:72: W291 trailing whitespace
#13 2.285 ./python_on_whales/components/compose/cli_wrapper.py:270:25: W291 trailing whitespace
```
I believe this is the reason

**Comment by anesabml** [comment]
I fixed those lint issues but the ci still fails on my local machine.

**Comment by gabrieldemarmiesse** [comment]
We use the balck formatter on this codebase, this is why the ci is complaining.

Could you do `isort ./ && black ./` and push again? This should fix those formatting problems. Thanks!

**Comment by anesabml** [comment]
@gabrieldemarmiesse  I run the test I added on my local machine (MacBook Pro), any ideas why it fails in CI?

**Comment by gabrieldemarmiesse** [comment]
I will take a look today, thanks again for the pull request!

**Comment by gabrieldemarmiesse** [comment]
I see the problem, but I don't know the root cause.
In the CI, during the test, we see:
```
alpine_1  | 5 packets transmitted, 0 packets received, 100% packet loss
```

locally, I have 
```
alpine_1  | 5 packets transmitted, 5 packets received, 0% packet loss
```

I wonder if we're just unlucky and that's a temporary network issue in the CI, I'll restart the job.

**Comment by gabrieldemarmiesse** [comment]
It's not luck it seems. I'll try to find out how we can fix this test.

File: python_on_whales/components/compose/cli_wrapper.py [hunk 1]
```diff
@@ -265,6 +271,9 @@ def up(self, services: List[str] = [], build: bool = False, detach: bool = False
                 exist.
             detach: If `True`, run the containers in the background. If `False` this
                 function returns only when all containers have stopped.
+                Incompatible with --abort-on-container-exit.
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/compose/cli_wrapper.py:274]:
```suggestion
                Incompatible with `abort_on_container_exit=True`.
```

File: python_on_whales/components/compose/cli_wrapper.py [hunk 2]
```diff
@@ -265,6 +271,9 @@ def up(self, services: List[str] = [], build: bool = False, detach: bool = False
...
+            abort_on_container_exit: If `True` stops all containers if any container was
+                stopped. Incompatible with --detach.
```
Comment (gabrieldemarmiesse) [inline python_on_whales/components/compose/cli_wrapper.py:276]:
```suggestion
                stopped. Incompatible with `detach=True`.
```

**Review by gabrieldemarmiesse — CHANGES_REQUESTED** [review]
Thank you for the pull request, just a few changes to make the docs better

**Comment by anesabml** [comment]
Thanks a lot for helping with debugging the issue, I will make the required changes regarding the docs.

File: python_on_whales/components/compose/cli_wrapper.py (hunk 1 above)
Comment (anesabml) [reply python_on_whales/components/compose/cli_wrapper.py:274]: That's better thanks for the suggestion

File: python_on_whales/components/compose/cli_wrapper.py (hunk 2 above)
Comment (anesabml) [reply python_on_whales/components/compose/cli_wrapper.py:276]: That's better thanks for the suggestion

**Review by gabrieldemarmiesse — APPROVED** [review]

**Comment by gabrieldemarmiesse** [comment]
It seems there is an issue on master, the CI doesn't pass suddently. Very likely something was updated and it caused the various issues we saw in your pull request. You can consider that this pull request is done (it was approuved). I'll work on the CI to make it work again. Thanks for your patience :)

**Comment by anesabml** [comment]
Thank you for your help.

**Pull request merged**